src/back/hlsl/writer.rs

```diff
@@ -1538,3 +1829,23 @@ fn sampling_str(sampling: crate::Sampling) -> Option<&'static str> {
         S::Sample => Some("sample"),
     }
 }
+
+fn storage_format_to_texture_type(format: crate::StorageFormat) -> Option<&'static str> {
+    use crate::StorageFormat as Sf;
+
+    match format {
+        Sf::Rgba8Unorm
+        | Sf::Rgba8Snorm
+        | Sf::Rgba16Float
+        | Sf::R32Float
+        | Sf::Rg32Float
+        | Sf::Rgba32Float => Some("float4"),
+        Sf::Rgba8Uint | Sf::Rgba16Uint | Sf::R32Uint | Sf::Rg32Uint | Sf::Rgba32Uint => {
+            Some("uint4")
+        }
+        Sf::Rgba8Sint | Sf::Rgba16Sint | Sf::R32Sint | Sf::Rg32Sint | Sf::Rgba32Sint => {
+            Some("int4")
+        }
+        _ => None,
```

Comment:
is this unfinished on purpose? are there any blockers or concerns here?

Reply:
Waiting for `Tint` to finish this :laughing:

https://dawn.googlesource.com/tint/+/refs/heads/main/src/writer/hlsl/generator_impl.cc#69

Reply:
Now we returns error here.